pkg/nats/stream/consumer_client.go

```diff
@@ -71,8 +71,9 @@ type ConsumerClientParams struct {
 
 // ConsumerClient represents a NATS streaming client.
 type ConsumerClient struct {
-	Conn *nats.Conn
-	mu   sync.RWMutex // Protects access to the response map.
+	Conn        *nats.Conn
+	mu          sync.RWMutex // Protects access to the response map.
+	muReqSubMap sync.RWMutex // Protects access to reqSubMap
```

Comment:
I think having two locks is an over-optimization here not worth the added complexity. Just using `mu` should be enough, as long as we use read locks when processing heartbeats? What do you think?